integration/fabric/iou/iou_test.go

```diff
@@ -40,42 +40,52 @@ var _ = Describe("EndToEnd", func() {
 		})
 
 		It("succeeded", func() {
-			res, err := ii.Client("borrower").CallView(
-				"create", common.JSONMarshall(&views.Create{
-					Amount:   10,
-					Lender:   ii.Identity("lender"),
-					Approver: ii.Identity("approver"),
-				}),
-			)
-			Expect(err).NotTo(HaveOccurred())
-			Expect(res).NotTo(BeNil())
-			id := common.JSONUnmarshalString(res)
-
-			res, err = ii.CLI("borrower").CallView("query", common.JSONMarshall(&views.Query{LinearID: id}))
-			Expect(err).NotTo(HaveOccurred())
-			Expect(common.JSONUnmarshalInt(res)).To(BeEquivalentTo(10))
-			res, err = ii.Client("lender").CallView("query", common.JSONMarshall(&views.Query{LinearID: id}))
-			Expect(err).NotTo(HaveOccurred())
-			Expect(common.JSONUnmarshalInt(res)).To(BeEquivalentTo(10))
+			iouState := CreateIOU(ii, "")
+			CheckState(ii, "borrower", iouState, 10)
+			CheckState(ii, "lender", iouState, 10)
+			UpdateIOU(ii, iouState, 5)
+			CheckState(ii, "borrower", iouState, 5)
+			CheckState(ii, "lender", iouState, 5)
 
-			Expect(id).NotTo(BeNil())
-			txIDBoxed, err := ii.Client("borrower").CallView(
-				"update", common.JSONMarshall(&views.Update{
-					LinearID: id,
-					Amount:   5,
-					Approver: ii.Identity("approver"),
-				}),
-			)
-			Expect(err).NotTo(HaveOccurred())
-			txID := common.JSONUnmarshalString(txIDBoxed)
-			Expect(ii.Client("lender").IsTxFinal(txID)).NotTo(HaveOccurred())
-
-			res, err = ii.Client("borrower").CallView("query", common.JSONMarshall(&views.Query{LinearID: id}))
-			Expect(err).NotTo(HaveOccurred())
-			Expect(common.JSONUnmarshalInt(res)).To(BeEquivalentTo(5))
-			res, err = ii.Client("lender").CallView("query", common.JSONMarshall(&views.Query{LinearID: id}))
-			Expect(err).NotTo(HaveOccurred())
-			Expect(common.JSONUnmarshalInt(res)).To(BeEquivalentTo(5))
+			iouState = CreateIOU(ii, "borrower-hsm-2")
+			CheckState(ii, "borrower", iouState, 10)
+			CheckState(ii, "lender", iouState, 10)
+			UpdateIOU(ii, iouState, 5)
+			CheckState(ii, "borrower", iouState, 5)
+			CheckState(ii, "lender", iouState, 5)
 		})
 	})
 })
+
+func CreateIOU(ii *integration.Infrastructure, identityLabel string) string {
```

Comment:
Maybe these methods can be private

Reply:
Added amount. For the visibility, given the split Marcus asked. These methods are now used in another package as well.